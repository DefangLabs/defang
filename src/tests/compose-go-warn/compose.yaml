name: compose-go-warning
services:
  echo:
    image: ealen/echo-server
    ports:
      - target: 80
        mode: ingress
secrets:
  dummy:
    external: yes # compose-go gives warning for using `yes` for boolean values
    name: dummy
