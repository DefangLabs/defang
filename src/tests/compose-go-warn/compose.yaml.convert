[
  {
    "name": "echo",
    "image": "ealen/echo-server",
    "platform": 2,
    "internal": true,
    "ports": [
      {
        "target": 80,
        "mode": 1
      }
    ],
    "networks": 1
  }
]