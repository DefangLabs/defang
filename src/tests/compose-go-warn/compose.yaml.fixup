[
  {
    "command": null,
    "entrypoint": null,
    "image": "ealen/echo-server",
    "networks": {
      "default": null
    },
    "ports": [
      {
        "mode": "ingress",
        "target": 80,
        "protocol": "tcp",
        "app_protocol": "http"
      }
    ]
  }
]