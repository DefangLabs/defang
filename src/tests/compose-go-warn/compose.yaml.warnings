 ! "yes" for boolean is not supported by YAML 1.2, please use `true`
 ! service "echo": ingress port without healthcheck defaults to GET / HTTP/1.1
 ! service "echo": missing compose directive: restart; assuming 'unless-stopped' (add 'restart' to silence)
 ! service "echo": missing memory reservation; using provider-specific defaults. Specify deploy.resources.reservations.memory to avoid out-of-memory errors