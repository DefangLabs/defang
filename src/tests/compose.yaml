name: tests
services:
  dfnx:
    build:
      context: .
      dockerfile: Dockerfile
      args:
        DNS: "${COMPOSE_PROJECT_NAME}-dfnx.prod1.defang.dev"
    env_file:
      - fileName.env
    deploy:
      resources:
        limits:
          cpus: '0.50'
          memory: 512M
        reservations:
          cpus: '0.25'
          memory: 256M
    ports:
      - target: 80
        mode: ingress
      - target: 1234
        # mode: host
    secrets:
      - dummy
    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost/"]
      # disable: true

  # dfnx:
  #   build:
  #     context: .
      # dockerfile: Dockerfile.dfn
    # ports:
      # - 80

  echo:
    image: ealen/echo-server
    ports:
      - target: 80
        mode: ingress
    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost/"]
    # domainname: echotest.gnafed.click
    profiles:
      - donotstart

secrets:
  dummy:
    external: true
    name: dummy
