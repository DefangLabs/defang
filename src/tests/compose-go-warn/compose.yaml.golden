name: test
services:
  echo:
    image: ealen/echo-server
    networks:
      default: null
    ports:
      - mode: ingress
        target: 80
networks:
  default:
    name: test_default
secrets:
  dummy:
    name: dummy
    external: true
